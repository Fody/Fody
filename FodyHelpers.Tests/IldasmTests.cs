using System;
using ApprovalTests;
using DummyAssembly;
using Fody;
using Xunit;
using Xunit.Abstractions;

public class IldasmTests :
    XunitApprovalBase
{
    IDisposable disposable;

    [Fact]
    public void StaticPathResolution()
    {
        Assert.True(Ildasm.FoundIldasm);
    }

    public IldasmTests(ITestOutputHelper outputHelper) :
        base(outputHelper)
    {
        disposable = RuntimeNamer.BuildForRuntimeAndConfig();
    }

    [Fact]
    public void VerifyMethod()
    {
        var verify = Ildasm.Decompile(GetAssemblyPath(), "DummyAssembly.Class1::Method");
        Approvals.Verify(verify);
    }

    [Fact]
    public void Verify()
    {
        var verify = Ildasm.Decompile(GetAssemblyPath());
        Approvals.Verify(verify);
    }

    static string GetAssemblyPath()
    {
        var assembly = typeof(Class1).Assembly;

        var uri = new UriBuilder(assembly.CodeBase);
        return Uri.UnescapeDataString(uri.Path);
    }

    public override void Dispose()
    {
        disposable.Dispose();
        base.Dispose();
    }
}