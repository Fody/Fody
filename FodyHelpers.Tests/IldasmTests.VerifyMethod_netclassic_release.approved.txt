.class public auto ansi beforefieldinit DummyAssembly.Class1
extends [mscorlib]System.Object
{
.method public hidebysig instance void
Method(int32 param) cil managed
{
.maxstack  8
IL_0000:  ldarg.0
IL_0001:  ldarg.1
IL_0002:  stfld      int32 DummyAssembly.Class1::'field'
IL_0007:  ldarg.1
IL_0008:  ldc.i4.1
IL_0009:  add
IL_000a:  starg.s    param
IL_000c:  ldarg.1
IL_000d:  ldarg.0
IL_000e:  ldfld      int32 DummyAssembly.Class1::'field'
IL_0013:  bne.un.s   IL_0023
IL_0015:  ldarg.0
IL_0016:  ldarg.0
IL_0017:  ldfld      int32 DummyAssembly.Class1::'field'
IL_001c:  ldc.i4.1
IL_001d:  sub
IL_001e:  stfld      int32 DummyAssembly.Class1::'field'
IL_0023:  ret
}
}
